determined-ai/determined
# fix: allow deletion of workspaces when case-insensitive matches exist

## Description

Addresses DET-9783 by adding a new case-sensitive name search parameter to the `GetWorkspaces` endpoint, and then uses that for the lookup of workspace when deleting by name on the CLI.

A (potentially undesirable?) consequence of this is that case-insensitive deletion by name will stop working, but that doesn't seem like it was intentional to me, anyway. If this is desired, it's achievable, just let me know.


## Test Plan

1. Create two workspaces whose names are different but match when compared case-insensitive (e.g. `foobar` and `FooBar`).
2. Using the new CLI code (might need to `make -C harness install`), delete one of them (e.g. `det w delete foobar`) and confirm deletion.
3. Verify that you get a success message (instead of a message like "AssertionError: workspace name is assumed to be unique. Failed to delete workspace") and make double sure by listing workspaces with `det w` and verifying the deleted workspace doesn't show up in the listing.
4. Clean up by also deleting the second workspace you created (e.g. `det w delete FooBar`)


## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
DET-9783
